rviz_common/src/rviz_common/frame_manager.cpp

```diff
@@ -148,13 +152,13 @@ void FrameManager::syncTime(rclcpp::Time time)
       sync_time_ = time;
       break;
     case SyncApprox:
-      if (time == rclcpp::Time(0)) {
+      if (time == rclcpp::Time(0, 0, clock_->get_clock_type())) {
```

Comment:
I added the clock type to all Time constructors because this condition failed when not passing the type:
https://github.com/ros2/rviz/blob/5bda5bf55e78d953bca946605aa523c39af086a9/rviz_common/src/rviz_common/frame_manager.cpp#L174-L179

The default clock type for the `Time(microsecond)` constructor is `RCL_SYSTEM_TIME`, but the time passed into the upper method is implicitly created from a message using this constructor:
[ros2/rclcpp/time.hpp:43](https://github.com/ros2/rclcpp/blob/5e64191e10b4ecc46a8e2a5d4744cceeade49333/rclcpp/include/rclcpp/time.hpp#L43-L46)

As a precaution I added the clock type to all Time instantiations.
It might not be necessary for all of them.


The `(second, nanosecond, type)` constructor was necessary because the C++ compiler complained about an ambiguous call when using `(nanosecond, type)`.